File: QytetetRuby/lib/tablero.rb
#encoding: utf-8
module ModeloQytetet


class Tablero
	attr_reader :casillas, :carcel
	
	def initialize
		@casillas = Array.new
		@carcel
		
		inicializar
	end
	
	def inicializar
		@casillas << Casilla.new(0, 1000, TipoCasilla::PARKING)
		@casillas << Calle.new(1, TituloPropiedad.new("Calle1", 500, 50, 0.13, 150, 250))
		@casillas << Calle.new(2, TituloPropiedad.new("Calle2", 550, 50,  0.12, 200, 260))
		@casillas << Calle.new(3, TituloPropiedad.new("Calle3", 520, 55,  0.11, 180, 250))
		@casillas << Casilla.new(4, 0, TipoCasilla::SORPRESA)
		
		@carcel =  Casilla.new(5, 0, TipoCasilla::CARCEL)
		@casillas << @carcel
		@casillas << Calle.new(6, TituloPropiedad.new("Calle4", 720, 70,  0.13, 450, 360))
		@casillas << Calle.new(7, TituloPropiedad.new("Calle5", 800, 80,  0.12, 500, 400))
		@casillas << Casilla.new(8, -200, TipoCasilla::IMPUESTO)
		@casillas << Calle.new(9, TituloPropiedad.new("Calle6", 750, 75,  0.15, 500, 420))
		
		@casillas << Calle.new(10, TituloPropiedad.new("Calle7", 600, 95,  0.14, 700, 425))
		@casillas << Casilla.new(11, 0, TipoCasilla::SORPRESA)
		@casillas << Calle.new(12, TituloPropiedad.new("Calle8", 560, 56,  0.17, 700, 425))
		@casillas << Calle.new(13, TituloPropiedad.new("Calle9", 700, 70,  0.15, 700, 425))
		@casillas << Calle.new(14, TituloPropiedad.new("Calle10", 580, 60,  0.13, 700, 425))
		
		@casillas << Casilla.new(15, 0, TipoCasilla::JUEZ)		
		@casillas << Calle.new(16, TituloPropiedad.new("Calle11", 925, 95,  0.18, 700, 425))
		@casillas << Calle.new(17, TituloPropiedad.new("Calle12", 950, 95,  0.19, 750, 450))
		@casillas << Casilla.new(18, 0, TipoCasilla::SORPRESA)
		@casillas << Calle.new(19, TituloPropiedad.new("Calle13", 1000, 100,  0.2, 1000, 450))
	end
	
	def esCasillaCarcel(numeroCasilla)
		numeroCasilla == @carcel.numero
	end
	
	def obtenerCasillaFinal(casilla, desplazamiento)
		obtenerCasillaNumero((casilla.numero + desplazamiento) % 20)
	end
	
	#Precondicion: numeroCasilla > 0 y menor que el max
	def obtenerCasillaNumero(numeroCasilla)
		@casillas[numeroCasilla]
	end
	
	def to_s()
		"Tablero{casillas=#{@casillas}, carcel=#{@carcel}}"
	end
	
	private :inicializar
end
end

File: QytetetJava/src/modeloqytetet/Tablero.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

import java.util.ArrayList;

/**
 *
 * @author jakerxd
 */
public class Tablero {
	private ArrayList<Casilla> casillas;
	private Casilla carcel;

	public Tablero()
	{
		inicializar();
	}
	
	ArrayList<Casilla> getCasillas() {
		return casillas;
	}

	Casilla getCarcel() {
		return carcel;
	}
	
	// precondicion: numeroCasilla > 0 && numeroCasilla < numeroMaximo
	Casilla obtenerCasillaNumero(int numeroCasilla)
	{
		return casillas.get(numeroCasilla);
	}
	
	Casilla obtenerCasillaFinal(Casilla casilla, int desplazamiento)
	{
		return obtenerCasillaNumero((casilla.getNumeroCasilla() + desplazamiento) % 20);
	}
	
	boolean esCasillaCarcel(int numeroCasilla)
	{
		return numeroCasilla == carcel.getNumeroCasilla();
	}
	
	private void inicializar(){
		casillas=new ArrayList();
		
		casillas.add(new OtraCasilla(0, 1000, TipoCasilla.PARKING));
		casillas.add(new Calle(1, new TituloPropiedad("Calle1", 500, 50, (float) 0.13, 150, 250)));
		casillas.add(new Calle(2, new TituloPropiedad("Calle2", 550, 50, (float) 0.12, 200, 260)));
		casillas.add(new Calle(3, new TituloPropiedad("Calle3", 520, 55, (float) 0.11, 180, 250)));
		casillas.add(new OtraCasilla(4, 0, TipoCasilla.SORPRESA));
		
		// Preguntar si se copia o se apunta
		carcel = new OtraCasilla(5, 0, TipoCasilla.CARCEL);
		casillas.add(carcel);
		casillas.add(new Calle(6, new TituloPropiedad("Calle4", 720, 70, (float) 0.13, 450, 360)));
		casillas.add(new Calle(7, new TituloPropiedad("Calle5", 800, 80, (float) 0.12, 500, 400)));
		casillas.add(new OtraCasilla(8, -200, TipoCasilla.IMPUESTO));
		casillas.add(new Calle(9, new TituloPropiedad("Calle6", 750, 75, (float) 0.15, 500, 420)));
		
		casillas.add(new Calle(10, new TituloPropiedad("Calle7", 600, 95, (float) 0.14, 700, 425)));
		casillas.add(new OtraCasilla(11, 0, TipoCasilla.SORPRESA));
		casillas.add(new Calle(12, new TituloPropiedad("Calle8", 560, 56, (float) 0.17, 700, 425)));
		casillas.add(new Calle(13, new TituloPropiedad("Calle9", 700, 70, (float) 0.15, 700, 425)));
		casillas.add(new Calle(14, new TituloPropiedad("Calle10", 580, 60, (float) 0.13, 700, 425)));
		
		casillas.add(new OtraCasilla(15, 0, TipoCasilla.JUEZ));		
		casillas.add(new Calle(16, new TituloPropiedad("Calle11", 925, 95, (float) 0.18, 700, 425)));
		casillas.add(new Calle(17, new TituloPropiedad("Calle12", 950, 95, (float) 0.19, 750, 450)));
		casillas.add(new OtraCasilla(18, 0, TipoCasilla.SORPRESA));
		casillas.add(new Calle(19, new TituloPropiedad("Calle13", 1000, 100, (float) 0.2, 1000, 450)));
	}
	
	@Override
	public String toString() {
		return "Tablero{" + "casillas=" + casillas.toString() + ", carcel=" + carcel.toString() + '}' + "\n";
	}
}
